### ISSUE TITLE
Rewrite in ES6

### ISSUE THREAD

**Issue by AugustinLF** [este/este-cli#37]
We agreed that it would be cool to update the CLI it to ES6, we talked about two ways:
- Build the sources before pushing to NPM (and use a dist/main.js as the main in package.json)
- Add a post-install step after NPM install

Iirc, you were more for the first one. I [figured](http://mammal.io/articles/using-es6-today/) that there is a `"prepublish": "npm run compile"` step in the package.json, which let's us work this way, in a transparent way. How do you feel about that?

**Comment by grabbou** [comment]
https://github.com/hharnisc/ot-p2p-model/blob/master/package.json#L20

**Comment by AugustinLF** [comment]
Yep, that's what I'm talking about (the `npm run compile` was just the example from this blog post).

**Comment by grabbou** [comment]
Yep, just sharing what I've got from twitter after asking what's the most `efficient` way :)

**Comment by AugustinLF** [comment]
Should we get rid of the `CoreObject`, since we'll have real classes (not familliar with this lib, but it doesn't seem to do more, does it)? And any specific reason why you use `arguments`everywhere, instead of normal parameters (I personally really dislike the use of `arguments`?

**Comment by grabbou** [comment]
> And any specific reason why you use argumentseverywhere, instead of normal parameters (I personally really dislike the use of arguments?

Where in particular?

**Comment by grabbou** [comment]
```js
var args = _.dropRight(arguments, 1);
```
This is just the way to get the last argument passed to the function as it's an array of passed bits to a command.

**Comment by AugustinLF** [comment]
Yeah, I actually understood, there was also the fact that you use it in `Task.prototype.start`, but it actually make sense, if you want to pass the blueprint type first. Wouldn't it be cleaner to pass an opts object with the correct props then? Or use a spead argument, as avocated by [AirBnB](https://github.com/airbnb/javascript#7.6)?

**Comment by grabbou** [comment]
Updated my comment in the meantime.

**Comment by grabbou** [comment]
1. We don't have ES6 support yet, so let's refer to the previous guide
2. We can't user spread operator before the first one goes in, like below:
```
function(...args, self) {}
```

**Comment by AugustinLF** [comment]
Yeah yeah, I was obviously talking about how to do this port to ES6!

**Comment by grabbou** [comment]
Ah, all righty :D Lost the point, sorry. In terms of the CoreObject, the main purpose was to make it super easy to actually extend base blueprints and do some fancy inheritance out of the box. I've removed it all in all in few places (Blueprints) in favour of `_.assign`, but still, with ES6 classes are going to be better.

**Comment by grabbou** [comment]
Step 1 - fork to another branch and rewrite the whole `lib` with proper inheritance. Things to solve:
- Our custom elements should just decorate built-in AST objects so it's still possible to use `get` and other things mentioned somewhere as far as I remember. This needs a huge clean up
